src/server/api/controllers/user.controller.js

```diff
@@ -0,0 +1,91 @@
+const knex = require('../../config/db');
+const Error = require('../lib/utils/http-error');
+
+const getUserById = async (id, role) => {
+  try {
+    const eventData = await knex.raw(
+      ` SELECT ev.id AS eventId,  org.name AS organization,  ev.event_date AS eventDate,  ev.created_at AS eventTime,  WEEK(ev.created_at) as eventWeek  FROM  events_users AS eu, users AS us,  events_students AS es,  organizations AS org, events AS ev  WHERE  us.id = eu.users_id  AND es.student_id = eu.id  AND org.id = us.organization_id  AND ev.id = es.event_id  AND eu.users_id = ? `,
+      [id],
+    );
+    if (eventData[0].length === 0) {
+      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    }
+    const dataToBeFetched = await transformData(eventData, role);
+    return dataToBeFetched;
+  } catch (error) {
+    return error.message;
+  }
+};
```

Comment:
Seems like this endpoint return a lot of things, just checking in that this is getting the `user` as it seems to select a lot of things